lua/apisix.lua

```diff
@@ -435,12 +435,42 @@ end
 
 
 function _M.http_header_filter_phase()
-    run_plugin("header_filter")
+    local api_ctx = ngx.ctx.api_ctx
+    if not api_ctx then
+        return
+    end
+
+    if router.global_rules and router.global_rules.values
+            and #router.global_rules.values > 0
+    then
+        local plugins = core.tablepool.fetch("plugins", 32, 0)
+        for _, global_rule in ipairs(router.global_rules.values) do
+            core.table.clear(plugins)
+            plugins = plugin.filter(global_rule, plugins)
+            run_plugin("header_filter", plugins, api_ctx)
+        end
```

Comment:
missing release `plugins` to tablepool

https://github.com/apache/incubator-apisix/pull/1364/files#diff-1be8f7892798606be4abee3b0bc37860R492